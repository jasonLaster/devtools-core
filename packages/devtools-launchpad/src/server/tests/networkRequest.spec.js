const { handleNetworkRequest } = require("../networkRequest");

describe("networkRequest", () => {
  it("inline base64 json", () => {
    const url =
      "data:application/json;base64,eyJmaWxlIjoibmc6Ly8vU2V0dGluZ3NNb2R1bGUvWm1iU2V0dGluZ3NPcmRlckVtYWlsTmcyLm5nZmFjdG9yeS5qcyIsInZlcnNpb24iOjMsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIm5nOi8vL1NldHRpbmdzTW9kdWxlL1ptYlNldHRpbmdzT3JkZXJFbWFpbE5nMi5uZ2ZhY3RvcnkuanMiLCJuZzovLy9TZXR0aW5nc01vZHVsZS9abWJTZXR0aW5nc09yZGVyRW1haWxOZzIuaHRtbCJdLCJzb3VyY2VzQ29udGVudCI6WyIgIiwiPHptYi1zZXR0aW5ncy1vcmRlci1lbWFpbD48L3ptYi1zZXR0aW5ncy1vcmRlci1lbWFpbD4iXSwibWFwcGluZ3MiOiJBQUFBOzs7Ozs7MEJDQUE7TUFBQSw4QkFBQTtNQUFBO0lBQUE7OzsifQ==";

    const json = jest.fn();
    handleNetworkRequest({ query: { url } }, { json });
    expect(json.mock.calls[0]).toMatchSnapshot();
  });
});
