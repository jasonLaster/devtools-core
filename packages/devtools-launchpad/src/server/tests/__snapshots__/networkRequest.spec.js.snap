// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`networkRequest inline base64 json 1`] = `
Array [
  Object {
    "file": "ng:///SettingsModule/ZmbSettingsOrderEmailNg2.ngfactory.js",
    "mappings": "AAAA;;;;;;0BCAA;MAAA,8BAAA;MAAA;IAAA;;;",
    "sourceRoot": "",
    "sources": Array [
      "ng:///SettingsModule/ZmbSettingsOrderEmailNg2.ngfactory.js",
      "ng:///SettingsModule/ZmbSettingsOrderEmailNg2.html",
    ],
    "sourcesContent": Array [
      " ",
      "<zmb-settings-order-email></zmb-settings-order-email>",
    ],
    "version": 3,
  },
]
`;
